File: api/editor/api-editor/src/main/java/ca/bc/gov/tno/areas/editor/models/ScheduleModel.java
package ca.bc.gov.tno.areas.editor.models;

import java.time.LocalTime;
import java.time.ZonedDateTime;
import java.util.EnumSet;

import ca.bc.gov.tno.dal.db.Months;
import ca.bc.gov.tno.dal.db.ScheduleType;
import ca.bc.gov.tno.dal.db.WeekDays;
import ca.bc.gov.tno.dal.db.entities.Schedule;
import ca.bc.gov.tno.models.AuditColumnModel;

public class ScheduleModel extends AuditColumnModel {
  /**
   * Primary key to identify the schedule.
   */
  private int id;

  /**
   * A unique name to identify the schedule.
   */
  private String name;

  /**
   * A description of the schedule.
   */
  private String description = "";

  /**
   * Whether this record is enabled or disabled.
   */
  private boolean enabled = true;

  /**
   * The schedule type.
   */
  private ScheduleType scheduleType = ScheduleType.Repeating;

  /**
   * The number of milliseconds the service should rest before running again.
   */
  private int delayMS;

  /**
   * The date and time the service should begin running on. This is useful if a
   * service should be delayed from running for a period of time.
   */
  private ZonedDateTime runOn;

  /**
   * The time to start.
   */
  private LocalTime startAt;

  /**
   * The time to stop.
   */
  private LocalTime stopAt;

  /**
   * Number of times to run before waiting for next RunAt. "0" is used for
   * continuous running.
   */
  private int repeat;

  /**
   * Identify which week days the service should run.
   */
  private EnumSet<WeekDays> runOnWeekDays;

  /**
   * Identify which months the service should run.
   */
  private EnumSet<Months> runOnMonths;

  /**
   * Identify the day of the month the service should run.
   */
  private int dayOfMonth;

  /**
   * A collection of data sources that belong to this schedule.
   */
  // private List<DataSource> data_sources = new ArrayList<>();

  public ScheduleModel() {
  }

  public ScheduleModel(Schedule entity) {
    super(entity);

    if (entity != null) {
      this.id = entity.getId();
      this.name = entity.getName();
      this.description = entity.getDescription();
      this.enabled = entity.isEnabled();
      this.scheduleType = entity.getScheduleType();
      this.delayMS = entity.getDelayMS();
      this.runOn = entity.getRunOn();
      this.startAt = entity.getStartAt();
      this.stopAt = entity.getStopAt();
      this.repeat = entity.getRepeat();
      this.runOnWeekDays = entity.getRunOnWeekDays();
      this.runOnMonths = entity.getRunOnMonths();
      this.dayOfMonth = entity.getDayOfMonth();
    }
  }

  /**
   * @return int return the id
   */
  public int getId() {
    return id;
  }

  /**
   * @param id the id to set
   */
  public void setId(int id) {
    this.id = id;
  }

  /**
   * @return String return the name
   */
  public String getName() {
    return name;
  }

  /**
   * @param name the name to set
   */
  public void setName(String name) {
    this.name = name;
  }

  /**
   * @return String return the description
   */
  public String getDescription() {
    return description;
  }

  /**
   * @param description the description to set
   */
  public void setDescription(String description) {
    this.description = description;
  }

  /**
   * @return boolean return the enabled
   */
  public boolean getEnabled() {
    return enabled;
  }

  /**
   * @param enabled the enabled to set
   */
  public void isEnabled(boolean enabled) {
    this.enabled = enabled;
  }

  /**
   * @return int return the delayMS
   */
  public int getDelayMS() {
    return delayMS;
  }

  /**
   * @param delayMS the delayMS to set
   */
  public void setDelayMS(int delayMS) {
    this.delayMS = delayMS;
  }

  /**
   * @return ZonedDateTime return the runOn
   */
  public ZonedDateTime getRunOn() {
    return runOn;
  }

  /**
   * @param runOn the runOn to set
   */
  public void setRunOn(ZonedDateTime runOn) {
    this.runOn = runOn;
  }

  /**
   * @return int return the repeat
   */
  public int getRepeat() {
    return repeat;
  }

  /**
   * @param repeat the repeat to set
   */
  public void setRepeat(int repeat) {
    this.repeat = repeat;
  }

  /**
   * @return EnumSet{WeekDays} return the runOnWeekDays
   */
  public EnumSet<WeekDays> getRunOnWeekDays() {
    return runOnWeekDays;
  }

  /**
   * @param runOnWeekDays the runOnWeekDays to set
   */
  public void setRunOnWeekDays(EnumSet<WeekDays> runOnWeekDays) {
    this.runOnWeekDays = runOnWeekDays;
  }

  /**
   * @return EnumSet{Months} return the runOnMonths
   */
  public EnumSet<Months> getRunOnMonths() {
    return runOnMonths;
  }

  /**
   * @param runOnMonths the runOnMonths to set
   */
  public void setRunOnMonths(EnumSet<Months> runOnMonths) {
    this.runOnMonths = runOnMonths;
  }

  /**
   * @return int return the dayOfMonth
   */
  public int getDayOfMonth() {
    return dayOfMonth;
  }

  /**
   * @param dayOfMonth the dayOfMonth to set
   */
  public void setDayOfMonth(int dayOfMonth) {
    this.dayOfMonth = dayOfMonth;
  }

  /**
   * @return ScheduleType return the scheduleType
   */
  public ScheduleType getScheduleType() {
    return scheduleType;
  }

  /**
   * @param scheduleType the scheduleType to set
   */
  public void setScheduleType(ScheduleType scheduleType) {
    this.scheduleType = scheduleType;
  }

  /**
   * @return LocalTime return the startAt
   */
  public LocalTime getStartAt() {
    return startAt;
  }

  /**
   * @param startAt the startAt to set
   */
  public void setStartAt(LocalTime startAt) {
    this.startAt = startAt;
  }

  /**
   * @return LocalTime return the stopAt
   */
  public LocalTime getStopAt() {
    return stopAt;
  }

  /**
   * @param stopAt the stopAt to set
   */
  public void setStopAt(LocalTime stopAt) {
    this.stopAt = stopAt;
  }

}

File: libs/net/models/Areas/Admin/Ingest/ScheduleModel.cs
using TNO.API.Models;
using TNO.Entities;

namespace TNO.API.Areas.Admin.Models.Ingest;

/// <summary>
/// ScheduleModel class, provides a model that represents an schedule.
/// </summary>
public class ScheduleModel : AuditColumnsModel
{
    #region Properties
    /// <summary>
    /// get/set - Primary key to identify the schedule.
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// get/set -
    /// </summary>
    public string Name { get; set; } = "";

    /// <summary>
    /// get/set -
    /// </summary>
    public string Description { get; set; } = "";

    /// <summary>
    /// get/set -
    /// </summary>
    public bool IsEnabled { get; set; }

    /// <summary>
    /// get/set -
    /// </summary>
    public ScheduleType ScheduleType { get; set; } = ScheduleType.Continuous;

    /// <summary>
    /// get/set -
    /// </summary>
    public int DelayMS { get; set; }

    /// <summary>
    /// get/set -
    /// </summary>
    public DateTime? RunOn { get; set; }

    /// <summary>
    /// get/set -
    /// </summary>
    public TimeSpan? StartAt { get; set; }

    /// <summary>
    /// get/set -
    /// </summary>
    public TimeSpan? StopAt { get; set; }

    /// <summary>
    /// get/set -
    /// </summary>
    public bool RunOnlyOnce { get; set; }

    /// <summary>
    /// get/set -
    /// </summary>
    public int Repeat { get; set; }

    /// <summary>
    /// get/set -
    /// </summary>
    public ScheduleWeekDay RunOnWeekDays { get; set; }

    /// <summary>
    /// get/set -
    /// </summary>
    public ScheduleMonth RunOnMonths { get; set; }

    /// <summary>
    /// get/set -
    /// </summary>
    public int DayOfMonth { get; set; }

    /// <summary>
    /// get/set -
    /// </summary>
    public int? RequestedById { get; set; }
    #endregion

    #region Constructors
    /// <summary>
    /// Creates a new instance of an ScheduleModel.
    /// </summary>
    public ScheduleModel() { }

    /// <summary>
    /// Creates a new instance of an ScheduleModel, initializes with specified parameter.
    /// </summary>
    /// <param name="entity"></param>
    public ScheduleModel(Schedule? entity) : base(entity)
    {
        if (entity != null)
        {
            this.Id = entity.Id;
            this.Name = entity.Name;
            this.Description = entity.Description;
            this.IsEnabled = entity.IsEnabled;
            this.ScheduleType = entity.ScheduleType;
            this.DelayMS = entity.DelayMS;
            this.RunOn = entity.RunOn;
            this.StartAt = entity.StartAt;
            this.StopAt = entity.StopAt;
            this.RunOnlyOnce = entity.RunOnlyOnce;
            this.Repeat = entity.Repeat;
            this.RunOnWeekDays = entity.RunOnWeekDays;
            this.RunOnMonths = entity.RunOnMonths;
            this.DayOfMonth = entity.DayOfMonth;
            this.RequestedById = entity.RequestedById;
        }
    }
    #endregion
}
